pkg/services/grpcserver/query.go

```diff
@@ -54,112 +52,26 @@ func (s *queryDataServer) QueryData(ctx context.Context, in *pluginv2.QueryDataR
 	user := signedInUserFromBackendUser(req.PluginContext.User)
 	user.OrgID = req.PluginContext.OrgID
 
-	// split incoming queries into new QueryDataRequests based on the datasourceUID
-	requests, err := s.buildRequests(ctx, user, req)
-	if err != nil {
-		return nil, err
-	}
-
-	response := backend.NewQueryDataResponse()
-
-	// execute each request and merge the backend.DataResponse into a single QueryDataResponse
-	for _, req := range requests {
-		res, err := s.pluginClient.QueryData(ctx, req)
-		if err != nil {
-			return nil, err
-		}
-		for refID, res := range res.Responses {
-			response.Responses[refID] = res
-		}
-	}
-
-	return backend.ToProto().QueryDataResponse(response)
-}
-
-// TODO: can this be shared with the query service?
-func (s *queryDataServer) getDataSourceFromQuery(ctx context.Context, signedInUser *user.SignedInUser, skipCache bool, query *simplejson.Json) (*datasources.DataSource, error) {
-	uid := query.Get("datasource").Get("uid").MustString()
-
-	// before 8.3 special types could be sent as datasource (expr)
-	if uid == "" {
-		uid = query.Get("datasource").MustString()
-	}
-
-	if expr.IsDataSource(uid) {
-		return expr.DataSourceModel(), nil
-	}
-
-	if uid == grafanads.DatasourceUID {
-		return grafanads.DataSourceModel(signedInUser.OrgID), nil
-	}
-
-	// use datasourceId if it exists
-	id := query.Get("datasourceId").MustInt64(0)
-	if id > 0 {
-		ds, err := s.dataSourceCache.GetDatasource(ctx, id, signedInUser, skipCache)
-		if err != nil {
-			return nil, err
-		}
-		return ds, nil
+	metricReq := dtos.MetricRequest{
+		From:    strconv.FormatInt(req.Queries[0].TimeRange.From.UnixMilli(), 10),
+		To:      strconv.FormatInt(req.Queries[0].TimeRange.To.UnixMilli(), 10),
+		Queries: make([]*simplejson.Json, 0),
```

Comment:
In this approach we lose support for mixed time ranges 🤔

Reply:
good call. :thinking: i could split them into separate `MetricRequest`s, but that means that they would be split again by data source in `QueryDataMultipleSources`, which seems less efficient